source/configy/Read.d

```diff
@@ -0,0 +1,1928 @@
+            dbgWrite("Field name `%s` will use YAML field `%s`",
+                     FR.FieldName.paint(Yellow), FR.Name.paint(Green));
+
+
+        // Using exact type here matters: we could get a qualified type
+        // (e.g. `immutable(string)`) if the field is qualified,
+        // which causes problems.
+        FR.Type default_ = __traits(getMember, defaultValue, FR.FieldName);
+
+        // If this struct is disabled, do not attempt to parse anything besides
+        // the `enabled` / `disabled` field.
+        if (!enabledState)
+        {
+            // Even this is too noisy
+            version (none)
+                dbgWrite("%s: %s field of disabled struct, default: %s",
+                         path.paint(Cyan), "Ignoring".paint(Yellow), default_);
+
+            static if (FR.Name == "enabled")
+                return false;
+            else static if (FR.Name == "disabled")
+                return true;
+            else
+                return default_;
+        }
+
+        if (auto ptr = FR.FieldName in fieldDefaults)
+        {
+            dbgWrite("Found %s (%s.%s) in `fieldDefaults",
+                     FR.Name.paint(Cyan), path.paint(Cyan), FR.FieldName.paint(Cyan));
+
+            if (ctx.strict && FR.FieldName in node)
+                throw new ConfigExceptionImpl("'Key' field is specified twice", path, FR.FieldName, node.startMark());
+            return (*ptr).parseFieldImpl!(FR)(path.addPath(FR.FieldName), default_, ctx)
+                .dbgWriteRet("Using value '%s' from fieldDefaults for field '%s'",
+                             FR.FieldName.paint(Cyan));
+        }
+
+        // This, `FR.Pattern`, and the field in `@Name` are special support for `dub`
+        static if (FR.Pattern)
+        {
+            static if (is(FR.Type : V[K], K, V))
+            {
+                static struct AAFieldRef
+                {
+                    ///
+                    private enum Ref = V.init;
+                    ///
+                    private alias Type = V;
+                }
+
+                static assert(is(K : string), "Key type should be string-like");
+            }
+            else
+                static assert(0, "Cannot have pattern on non-AA field");
+
+            AAFieldRef.Type[string] result;
+            foreach (pair; node.mapping)
+            {
+                const key = pair.key.as!string;
+                if (!key.startsWith(FR.Name))
+                    continue;
+                string suffix = key[FR.Name.length .. $];
+                if (suffix.length)
+                {
+                    if (suffix[0] == '-') {} //suffix = suffix[1 .. $];
+                    else continue;
+                }
+
+                result[suffix] = pair.value.parseFieldImpl!(AAFieldRef)(
+                    path.addPath(key), default_.get(key, AAFieldRef.Type.init), ctx);
+            }
+            bool hack = true;
+            if (hack) return result;
```

Comment:
Because if that `static if` is true, we *always* return at this point, and then the compiler complains that everything after is not reachable. But wrapping the rest of the function in a `else static if` makes the code much more ugly and the dub-specific change more intrusive.